### ISSUE TITLE
Does nbviewer work as a hub service with JupyterHub 2.x??

### ISSUE THREAD

**Issue by rcthomas** [jupyter/nbviewer#994]
I haven't tested extensively but I think that for nbviewer to work with JupyterHub 2.x requires that it switch from purely token auth (`HubAuth`) to use the hub as an OAuth provider (`HubOAuth`)?  I think nbviewer is maybe old enough that it doesn't explicitly use `HubAuthenticated` stuff, I don't see it in the code as such.  Let me know if this is mistaken!  Thanks.

**Comment by minrk** [comment]
Yes, I believe that's accurate. I think just switching to HubOAuth ought to be enough.
